### PR TITLE
refactor: Robustify migration 135

### PR DESCRIPTION
**Pull request by rekmarks**
Robustifies migration 135 by using more of our own types throughout the migration file. The result of reviewing migration 135 and ensuring that it's functioning correctly. The only behavioral difference is that `getChainIdForNetworkClientId()` will capture an exception if no chain id can be found, as opposed to this occurring outside of that function. The relevant test cases have been updated to ensure that we capture this exception.

### CODE REVIEW

File: app/scripts/migrations/135.ts [hunk 1]
```diff
@@ -1,41 +1,58 @@
...
+const PermissionNames = {
+  eth_accounts: 'eth_accounts',
+  permittedChains: 'endowment:permitted-chains',
+} as const;
 
-type VersionedData = {
-  meta: { version: number };
-  data: Record<string, unknown>;
-};
+const BUILT_IN_NETWORKS: ReadonlyMap<string, string> = new Map([
```
Comment (jiexi) [inline app/scripts/migrations/135.ts:46]: thoughts on using `Hex` here to the value type?

**Review by jiexi — APPROVED** [review]
Small nit. I don't think this is blocked by approval since it's not merging into `main` so feel free to merge if you feel the nit worth addressing

File: app/scripts/migrations/135.ts (hunk 1 above)
Comment (adonesky1) [reply app/scripts/migrations/135.ts:46]: `goerli` still exists in state by this migration?

Comment (rekmarks) [reply app/scripts/migrations/135.ts]: Hex works for the values but not the keys

Comment (rekmarks) [reply app/scripts/migrations/135.ts]: Or you mean `goerli` exists as a value?

**Pull request merged**